### PR TITLE
release vedirect & Tado to latest

### PR DESCRIPTION
**Pull request by DutchmanNL**
- release vedirect
- npm package ready this evening
- testing OK
- JS 2.0 compatible

### CODE REVIEW

File: sources-dist.json [hunk 1]
```diff
@@ -1247,6 +1247,11 @@
     "icon": "https://raw.githubusercontent.com/hometm/ioBroker.vcard/master/admin/vcard.png",
     "type": "misc-data"
   },
+  "vedirect": {
+    "meta": "https://raw.githubusercontent.com/hometm/iobroker-community-adapters.vedirect/master/io-package.json",
```
Comment (ldittmar81) [inline sources-dist.json]: Wrong link

Comment (DutchmanNL) [reply sources-dist.json]: wtf happend, solved

Comment (DutchmanNL) [reply sources-dist.json]: and another typo fixed

**Comment by GermanBluefox** [comment]
image

**Comment by GermanBluefox** [comment]
And README must be cleared

**Comment by DutchmanNL** [comment]
fixed

**Comment by Apollon77** [comment]
Now we stay left with:

[E140] width and height of logo are not equal

as error

**Comment by Apollon77** [comment]
Some review comments:
* Blocker: Do not use setObject because this would overwrite history settings! Please use setObjectNotExists
* All data with "role: 'state'" feels not optimal. Can you not define better roles for certain states?
* I think some serialpoert error management (listening to error events) would be helpfull, else adapter will crash when an error happens here
* setting connection state to true on EACH received data package is not ideal. Please more do ion "On(connect)" or such
* your "ID to text mappings" could be done with better performance as an ddefined object where you simply use the ID as key to lookup.
* in unload you also have code for TCP which is never initialized in ready or elsewhere
* you never set info.connection to false ...

**Comment by DutchmanNL** [comment]
remarks fixed, can you have another look please :)

**Comment by DutchmanNL** [comment]
added release. of Tado to this pull request

**Comment by Apollon77** [comment]
Will have tonight

**Comment by DutchmanNL** [comment]
All fixed for both adapters can you please have another look 🙂

**Comment by Apollon77** [comment]
Tonight if 2.0.34 works out as expected ;-)

**Comment by Apollon77** [comment]
Comments:
vedirect: 
* you still do not reset "info.connection" on serial disconnect, also errors are only logged, so on serial disconnect no reties are done and so not auto-fix try
* serialPort variable is wrong ... port is correct or ?! ... this can not work ... blocker?!

Tado:
* logging in warn?
* https://github.com/iobroker-community-adapters/ioBroker.tado/blob/master/main.js#L175 ?? :-) ... system.adapter. ...

Rest ok

**Comment by DutchmanNL** [comment]
> * you still do not reset "info.connection" on serial disconnect, also errors are only logged, so on serial disconnect no reties are done and so not auto-fix try
> * serialPort variable is wrong ... port is correct or ?! ... this can not work ... blocker?!

Fixed second part, retry is a good suggestion will look into that today


> * logging in warn?
> * https://github.com/iobroker-community-adapters/ioBroker.tado/blob/master/main.js#L175 ?? :-) ... system.adapter. ...

- about logging currently yes, will be changed before moving to stable 
- do you have a suggestion for the second one? I disable the adapter here if credentials are not provided

**Pull request merged**